src/orion/client/runner.py

```diff
@@ -27,6 +28,41 @@
 from orion.core.worker.trial import AlreadyReleased
 from orion.executor.base import AsyncException, AsyncResult
 
+log = logging.getLogger(__name__)
+
+
+class Protected(object):
```

Comment:
This context manager is great, but does it risk overridding this signal handler when using the cmdline interface? https://github.com/Epistimio/orion/blob/develop/src/orion/core/cli/hunt.py#L208

Reply:
Well yes, but it restores it after each scatter-gather loop

Reply:
so the signal handler will get called, simply later when everything is processed

Reply:
FYI https://docs.python.org/3/library/signal.html#signal.signal

> The previous signal handler will be returned

Reply:
check test_interrupted_scatter_gather_custom_signal

Reply:
Yeah I know how the signals handlers are overridden and can be restored, thanks. :) I missed that [part]( https://github.com/Epistimio/orion/pull/684/files/4446b76cabe152d59aff4b552b2ee517727552f5..ad715dc0f99d76b152719d43043cb4e1a90a2d54#diff-13cdef139faa8780b0667c70442b698de549631ba0ef27680325d56259270566R63) where you manually call the previous handler. That's good.

Also, the signal handler in orion.core.cli.hunt can be removed now, as the Protected context manager is already handling this properly.

Reply:
For reference, @Delaunay made me realize that the handler in hunt is slightly different because it raises KeyboardInterrupt. So we can keep it as is. I'm closing this discussion.

Thanks!